src/bacnet/bacdef.h

```diff
@@ -126,7 +126,7 @@
 #define BACNET_ARRAY_ALL 0xFFFFFFFFU
 /* For device object property references with no device id defined */
 #define BACNET_NO_DEV_ID   0xFFFFFFFFu
-#define BACNET_NO_DEV_TYPE 0xFFFFu
+#define BACNET_NO_DEV_TYPE ((BACNET_OBJECT_TYPE)0xFFFFu)
```

Comment:
I'm not 100% on whether adding `OBJECT_NONE` to `BACNET_OBJECT_TYPE` is the ideal solution, as it would add a constant that doesn't exist in the BACnet spec, but it's probably better than initializing everything to Analog Inputs.

Reply:
Maybe `MAX_BACNET_OBJECT_TYPE` could be used instead?

Reply:
There are other enums where this library creates a non-standard value:
```typedef enum {
    MIN_BINARY_PV = 0,  /* for validating incoming values */
    BINARY_INACTIVE = 0,
    BINARY_ACTIVE = 1,
    MAX_BINARY_PV = 1,  /* for validating incoming values */
    BINARY_NULL = 255   /* our homemade way of storing this info */
} BACNET_BINARY_PV;
```